llvm/lib/Transforms/InstCombine/InstCombineAddSub.cpp

```diff
@@ -2682,8 +2682,12 @@ static Instruction *foldFNegIntoConstant(Instruction &I, const DataLayout &DL) {
   // Fold negation into constant operand.
   // -(X * C) --> X * (-C)
   if (match(FNegOp, m_FMul(m_Value(X), m_Constant(C))))
-    if (Constant *NegC = ConstantFoldUnaryOpOperand(Instruction::FNeg, C, DL))
-      return BinaryOperator::CreateFMulFMF(X, NegC, &I);
+    if (Constant *NegC = ConstantFoldUnaryOpOperand(Instruction::FNeg, C, DL)) {
+      FastMathFlags Flag = I.getFastMathFlags() | FNegOp->getFastMathFlags();
+      Flag.setNoInfs(I.getFastMathFlags().noInfs() &&
+                     FNegOp->getFastMathFlags().noInfs());
```

Comment:
Avoid repeated calls to I.getFastMathFlags() and FNegOp->getFastMathFlags()?

Reply:
done